src/api/AuthService.ts

```diff
@@ -0,0 +1,17 @@
+import BaseServiceModel from '@/api/BaseServiceModel';
+
+class AuthService extends BaseServiceModel {
+  async login(data: { username: string; password: string; location?: string }) {
+    const apiResponse = await this.request.get('./api/login', { data });
```

Comment:
Perhaps it would be worth to think about a URL factory or sth? (not necessarily in this pr)

Reply:
yes, maybe for the future